### PR TITLE
feat(Attributes): add doc for attributes

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: pages/attributes.vue [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    <p
+      :class="paragraph"
+    >Attributes stand for properties you can add on OpenGeode objects, typically meshes. 
+    These properties are not directly stored on meshes, so you can imagine add Attributes
+    on other objects. </p>
+
+    <p
+      :class="paragraph"
+    >Attributes are managed by an 
+    <span v-html="attributeManager" />.
```
Comment (BotellaA) [inline pages/attributes.vue:21]: attributeManager or AttributeManager ? In all the file

File: pages/attributes.vue [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        The <strong>default value</strong> (here <code>12.3</code>): 
+        the second parameter of the method.
+      </li>
+    </ul>
+
+    <p
+      :class="paragraph"
+    >If the attribute already exists and has the same name, 
+    same storage and same value type, 
+    the returned pointer points on the pre-existing attribute
```
Comment (BotellaA) [inline pages/attributes.vue]: pointer points is not pretty

File: pages/attributes.vue [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    the modification will be operated on all associated attributes.
+    </p>
+
+  </v-layout>
+</template>
+
+<script>
+export default {
+  data: () => ({
+    section: 'display-2 ma-5 pa-5',
```
Comment (BotellaA) [inline pages/attributes.vue:331]: Not related to this PR but we should define a mixin to have the common data stored inside the same component.

File: pages/attributes.vue (hunk 1 above)
Comment (panquez) [reply pages/attributes.vue:21]: It appears as "AttributeManager" and link to the right page

File: pages/attributes.vue (hunk 2 above)
Comment (panquez) [reply pages/attributes.vue]:
```suggestion
   a pointer to the pre-existing attribute is returned
```
I suggest:

File: pages/attributes.vue [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+        the second parameter of the method.
+      </li>
+    </ul>
+
+    <p
+      :class="paragraph"
+    >If the attribute already exists and has the same name, 
+    same storage and same value type, 
+    the returned pointer points on the pre-existing attribute
+    without creating a new one. If the attribute name already exist but
```
Comment (panquez) [inline pages/attributes.vue]:
```suggestion
    without creating a new attribute. If the attribute name already exist but 
```

File: pages/attributes.vue (hunk 2 above)
Comment (BotellaA) [reply pages/attributes.vue]: The same formulation exists somewhere else in the file

File: pages/attributes.vue [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+    <code class="cpp primary--text">
+    const auto generic_attr = manager.find_generic_attribute( "first_attribute" );
+    const auto value = generic_attr->generic_value( 2 ); // get a float
+    </code>
+    <!-- eslint-enable -->
+
+    <p
+      :class="paragraph"
+    >
+    The pointer points to an
```
Comment (panquez) [inline pages/attributes.vue]:
```suggestion
    The pointed object is an
```

**Review by BotellaA — DISMISSED** [review]

File: pages/attributes.vue [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+    <p :class="paragraph">
+      In this guide, you will learn fundamental elements to understand and 
+      manipulate attributes in OpenGeode.
+    </p>
+
+    <h2 :class="section">What are attributes?</h2>
+
+    <p
+      :class="paragraph"
+    >Attributes stand for properties you can add on OpenGeode objects, typically meshes.
```
Comment (benjaminchauvin) [inline pages/attributes.vue]: add to?

File: pages/attributes.vue [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+      :class="paragraph"
+    >Attributes stand for properties you can add on OpenGeode objects, typically meshes. 
+    These properties are not directly stored on meshes, so you can imagine add Attributes
+    on other objects. </p>
+
+    <p
+      :class="paragraph"
+    >Attributes are managed by an 
+    <span v-html="attributeManager" />.
+    Meshes have manager defined on their vertices, edges, facets, and so on.
```
Comment (benjaminchauvin) [inline pages/attributes.vue]: a manager?

File: pages/attributes.vue [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+      :class="paragraph"
+    >The second line resizes the created
+    <span v-html="attributeManager" />.
+    It means that all Attributes associated to this 
+    <span v-html="attributeManager" /> will have exactly 10 values.
+    </p>
+
+    <p
+      :class="paragraph"
+    >Notice that this method can be called before and after
```
Comment (benjaminchauvin) [inline pages/attributes.vue]: I would say "Note" instead of "Notice". Maybe "Notice" works. Up to you.

File: pages/attributes.vue [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+      :class="paragraph"
+    >Notice that this method can be called before and after 
+    the creation of Attributes associated to the 
+    <span v-html="attributeManager" /> 
+    (as done at the fourth line of the above code). 
+    </p>
+
+    <p
+      :class="paragraph"
+    >As in standard library, there is also a method to reserve Attributes capacities.
```
Comment (benjaminchauvin) [inline pages/attributes.vue]: in the standard [...] Attribute capacities

**Review by benjaminchauvin — APPROVED** [review]

File: pages/attributes.vue [hunk 10]
```diff
@@ -0,0 +1,342 @@
...
+      manipulate attributes in OpenGeode.
+    </p>
+
+    <h2 :class="section">What are attributes?</h2>
+
+    <p
+      :class="paragraph"
+    >Attributes stand for properties you can add on OpenGeode objects, typically meshes. 
+    These properties are not directly stored on meshes, so you can imagine add Attributes
+    on other objects. </p>
```
Comment (BotellaA) [inline pages/attributes.vue]:
```suggestion
    >Attributes stand for properties you can add on OpenGeode objects. </p>
```
Fifi proposition

File: pages/attributes.vue [hunk 11]
```diff
@@ -0,0 +1,342 @@
...
+      :class="paragraph"
+    >Attributes are managed by an 
+    <span v-html="attributeManager" />.
+    Meshes have manager defined on their vertices, edges, facets, and so on.
+    Updating the mesh (e.g. by adding or removing vertices) automatically updates attached
+    <span v-html="attributeManager" />. </p>
+
+    <p
+      :class="paragraph"
+    >But you can create an
```
Comment (BotellaA) [inline pages/attributes.vue]:
```suggestion
    >You can create an
```
Fifi

File: pages/attributes.vue [hunk 12]
```diff
@@ -0,0 +1,342 @@
...
+      manager.resize( 10 );
+      auto attribute_ptr = manager.find_or_create< VariableAttribute, double >(
+        "first_attribute", 12.3 );
+      manager.resize( 6 );
+    </code>
+    <!-- eslint-enable -->
+
+    <p
+      :class="paragraph"
+    >The first line of the above code creates a default
```
Comment (BotellaA) [inline pages/attributes.vue:64]: Fifi: Is it better with line numbers written in the code?

File: pages/attributes.vue [hunk 13]
```diff
@@ -0,0 +1,342 @@
...
+
+    <p
+      :class="paragraph"
+    >First of all, let's get an  
+    <span v-html="attributeManager" />
+    by the dedicated mesh methods or by creating it:</p>
+
+    <!-- eslint-disable -->
+    <code class="cpp primary--text">
+      AttributeManager manager;
```
Comment (BotellaA) [inline pages/attributes.vue:54]: Add another example with the mesh attribute manager

File: pages/attributes.vue [hunk 14]
```diff
@@ -0,0 +1,342 @@
...
+        the first template parameter of the method (more details below);
+      </li>
+      <li>
+        The <strong>type</strong> of Attribute values (here <code>double</code>): 
+        the second template parameter of the method;
+      </li>
+      <li>
+        The <strong>default value</strong> (here <code>12.3</code>): 
+        the second parameter of the method.
+      </li>
```
Comment (BotellaA) [inline pages/attributes.vue]:
```suggestion
      <li>
        The <strong>type</strong> of Attribute storage (here a 
        <span v-html="variableAttribute" />): 
        the first template parameter of the method (more details below);
      </li>
      <li>
        The <strong>type</strong> of Attribute values (here <code>double</code>): 
        the second template parameter of the method;
      </li>
      <li>
        An <strong>attribute name</strong> (here <code>first_attribute</code>): 
        the first parameter of the method <code>find_or_create</code>;
      </li>
      <li>
        The <strong>default value</strong> (here <code>12.3</code>): 
        the second parameter of the method.
      </li>
```
Fifi (same order than the method)

File: pages/attributes.vue [hunk 15]
```diff
@@ -0,0 +1,342 @@
...
+      <li>
+        <span v-html="constantAttribute" />:  
+        all the attribute elements have the given default value as value.
+        Only one value is stored (the default value);
+      </li>
+      <li>
+        <span v-html="sparseAttribute" />:  
+        all the attribute elements have the given default value as value,
+        excepted the ones that have been explicitly defined.
+        The number of stored values is between one (only the default value)
```
Comment (BotellaA) [inline pages/attributes.vue:148]:
```suggestion
        The number of stored values is between one
```

File: pages/attributes.vue [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+      </li>
+      <li>
+        <span v-html="sparseAttribute" />:  
+        all the attribute elements have the given default value as value,
+        excepted the ones that have been explicitly defined.
+        The number of stored values is between one (only the default value) 
+        and the attribute size plus one 
+        (one value for each attribute element and the default value).
+        This storage is useful if the number of values different from the 
+        default value is small with regards to the attribute size.
```
Comment (BotellaA) [inline pages/attributes.vue:152]: Add a better idea on what "small" means: is it less than 50%, less than 90%??

**Comment by BotellaA** [comment]
Fifi: 

- Need for specific section for how to use attribute with meshes
- What happens if you delete an attribute value of an attribute on a mesh with deleting the mesh element

File: pages/attributes.vue (hunk 12 above)
Comment (panquez) [reply pages/attributes.vue:64]: Yes you are right, we will improve that.

File: pages/attributes.vue (hunk 16 above)
Comment (panquez) [reply pages/attributes.vue:152]: This kind of user is up to the user context. And I think there is some benchmarking to do to answer this question

**Comment by panquez** [comment]
@BotellaA  I do not understand your first point. For the second point it is possible to delete element from the AttributeManager from a mesh, it will also delete the mesh element. So your mesh will potentially not be valid anymore.

**Pull request merged**